src/main/java/com/pollistics/controllers/PollController.java

```diff
@@ -45,10 +55,30 @@ public String createPoll(HttpServletRequest request, Model model) {
 	}
 
 	@PostMapping(value = "/polls/vote/{pollId}")
-	public String voteOptions(@PathVariable String pollId, HttpServletRequest request, Model model) {
-		String option = request.getParameter("option");
-		Poll p = pollService.getPoll(pollId);
-		pollService.voteOption(p, option);
-		return "redirect:/" + pollId;
+	public String voteOptions(@CookieValue(value = "id", defaultValue = "") String cookieIdValue, @PathVariable String pollId, HttpServletRequest request, HttpServletResponse response, Model model) {
+		if (cookie.getValue().contains(pollId)) {
```

Comment:
Maybe we could seperate all this cookie logic into a seperate class or method later